text/0000-configadmin-ds-integration.md

```diff
@@ -0,0 +1,1752 @@
+// the caller until DS has finished it's processing. At this point in time DS has
+// finished creating the data structures for the factory instance and has registered
+// the factory instance. The state is SATISFIED. The component instance has not been
+// constructed yet because the immediate attribute is false. DS is waiting for someone
+// to request a service reference to it.
+ auto instance = GetInstance<test::CAInterface>();
+ 
+//The factory instance is now constructed.
+```
+After the factory instance is constructed in the previous code snippet, sample::VirtualFileSystem is 
+in the ACTIVE state. The constructor was called with
+an AnyMap input parameter containing:
+
+   - the component properties.
+       - component.name = sample::VirtualFileSystem~<random sequence generated by CreateFactoryConfiguration>
+       - component.id = unique id created by DS.
+       - cprop1 = "456"
+       - uniqueProp = "instance1"  
+cprop1 was included in the "properties" element of the  manifest.json file. uniqueProp was included in the Update 
+call that updated the properties for the factory instance. DS always adds component.name and component.id properties.
+
+The Service Registration properties for the factory instance in the framework have also been updated with the 
+same 4 properties.
+
+## Proposed Design Details - Configuration Admin
+### ListConfigurations
+ListConfigurations allows DS to look up configuration objects in the Configuration Admin repository. ListConfigurations 
+does not create an empty object if the requested configuration object does not exist (unlike GetConfiguration which 
+does create an empty object if one does not exist). 
+
+### Configuration Listener
+DS implements the configEvent on the ConfigurationListener interface to receive notifications from ConfigurationAdmin 
+when configuration objects are updated or deleted. Configuration Admin notifies all Configuration Listeners for all 
+configuration object changes. Each ConfigurationListener determines if the notification is for an object on which they 
+are listening and discards the rest.
+
+### Update, UpdateIfDifferent, Remove
+According to the Configuration Admin OSGI Specification notifications to listeners when configuration objects are 
+updated or removed must happen asynchronously. These three Configuration methods return before the asynchronous 
+operations were completed. Have changed the signature of these three methods to return a future instead of a void. 
+This gives the caller the opportunity to wait for the update methods to complete.
+
+## Alternate Designs Considered
+No alternate functional designs were considered.
+R4_Standardization (OSGi-compliance) is a must-have requirement, and by definition, we are following the functional 
+specifications described in the OSGi specification as it can be applied to C++.
+
+## Architectural Design 
+Declarative Services(DS) Component model is implemented as a layer on top of the OSGi Services model. A 
```

Comment:
DS is a service that layers on top of the Core Framework to extend the Core Framework and provide additional functionality. I'm confused by it being a layer on top of the OSGI Services model.